lib/sqlitex.ex

```diff
@@ -24,29 +24,46 @@ defmodule Sqlitex do
   {:ok, [%{a: 1, b: 2, c: 3}]}
 
   ```
+
+  ## Configuration
+
+  Sqlitex uses the Erlang library [esqlite](https://github.com/mmzeeman/esqlite)
+  which accepts a timeout parameter for almost all interactions with the database.
+  The default value for this timeout is 5000 ms. Many functions in Sqlitex accept
+  a timeout parameter that is passed on to the esqlite calls and that also defaults
+  to 5000 ms. If required, this default value can be overridden globally with the
+  following in your `config.exs`:
+
+  ```
+  config :sqlitex,
+    esqlite3_timeout: 10_000 # or other positive integer number of ms
+  ```
   """
 
-  @spec close(connection) :: :ok
-  def close(db) do
-    :esqlite3.close(db)
+  @esqlite3_timeout Application.get_env(:sqlitex, :esqlite3_timeout, 5_000)
```

Comment:
I could be wrong, but doesn't storing this in a module attribute mean that this gets read from the app config at compile time?  So if I update the `:esqlite3_timeout` in my config but don't explicitly recompile sqlitex then that change won't be picked up?

I guess maybe Elixir is smart enough to pick up that the change in sqlitex config might require a sqlitex recompile, but I'm not sure if that is the case...

Reply:
You know, I'm not sure either.

If we went with changing these to [use keyword lists anyway](https://github.com/mmmries/sqlitex/pull/65#discussion_r189240178) then the intended benefit of using module attributes (easy specification of default values in function definitions) is no longer needed.

Thnaks 👍